monkey/monkey_island/cc/ui/src/components/Main.tsx

```diff
@@ -80,43 +116,50 @@ class AppComponent extends AuthComponent {
     }
 
     if (res) {
+      // TODO: get plugins only one time
+      // TODO: Use real endpoint
+      if(!this.pluginsFetched) {
+        this.refreshAvailablePlugins();
+        this.refreshInstalledPlugins();
+          this.pluginsFetched = true;
+      }
+
       // update status: report generation
       this.authFetch('/api/report-generation-status', {}, false)
         .then(res => res.json())
         .then(res => {
           this.setState({
             completedSteps: new CompletedSteps(
-                                  this.state.completedSteps.runMonkey,
-                                  this.state.completedSteps.infectionDone,
-                                  res.report_done
-                                )
+              this.state.completedSteps.runMonkey,
```

Comment:
I think this is a worse formatting, but the actual solution is https://github.com/guardicore/monkey/issues/3070, no point debating  about it. But if you can avoid reformatting files, because we're on borrowed time and sorting what changes are irrelevant takes time